aries-site/src/pages/components/checkbox.js

```diff
@@ -24,11 +32,132 @@ const CheckBox = () => (
+          CheckBox can be used in the form of a toggle, a single option which
+          can indicate something such as on or off. Toggle is prefered to
+          quickly switch between two possible states. Toggles usually provides
+          immediate results in which gives users the ability to control their
+          preference.
+        </SubsectionText>
+        <SubsectionText>
+          CheckBox should not be used if a user should only be allowed to select
+          one option from a list of options. In the case that a user should only
+          be able to select one option, refer to{' '}
+          <Anchor href="/components/radiobutton">RadioButton</Anchor>
+        </SubsectionText>
+      </Subsection>
+      <Subsection name="About CheckBox" level={3} gap="small">
+        <SubsectionText>
+          There are three different states that CheckBox provides.
+          <BulletedList
+            level={3}
+            items={['Unchecked', 'Checked', 'Indeterminate']}
+          />
+          Checked and unchecked states refer to when an individual CheckBox is
+          checked or when all options from a group are checked. Indeterminate
+          can be used to indicate that a partial set of the available options
+          have been checked, this can be viewed in a form of a checkbox group.
+        </SubsectionText>
+      </Subsection>
+      <Subsection name="Accessibility" level={3} gap="small">
+        <SubsectionText>
+          CheckBox should be used inside of a FormField to ensure that a label
+          is properly associated with the input. The labels on individual
+          checkboxes should clearly describe the purpose of the checkbox.
+        </SubsectionText>
+        <SubsectionText>
+          If a CheckBox is used outside of the context of a FormField, it is
+          important to meet accessibility requirements in an alternate way.
+          There should always be clear visual indication, either by text or
+          icon, that describes the purpose of the checkbox.
+        </SubsectionText>
+      </Subsection>
+    </ContentSection>
+    <ContentSection>
+      <Subsection name="Variants">
+        <SubsectionText>
+          CheckBox can be used individually, within a group, or as a toggle. A
+          standalone or toggle checkbox indicates that a user is opting-in to
+          the context of the checkbox. Within a group, one or multiple
+          checkboxes can be selected.
+        </SubsectionText>
+      </Subsection>
+      <Subsection name="CheckBox with descriptions" level={3}>
+        <SubsectionText>
+          Adding a description provides the user additional information about
+          the context or purpose of the checkboxes.
+        </SubsectionText>
+        <Example
+          docs="https://v2.grommet.io/checkbox?theme=hpe#props"
+          code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/checkbox/CheckBoxDescription.js"
```

Comment:
Can we use consistent file naming conventions? Would like to see this one as 'CheckBoxDescriptionExample.js' (i.e. ComponentNameVariantNameExample.js)